deepmd/pt/entrypoints/main.py

```diff
@@ -114,9 +114,11 @@ def prepare_trainer_input_single(
             validation_dataset_params["systems"] if validation_dataset_params else None
         )
         training_systems = training_dataset_params["systems"]
-        training_systems = process_systems(training_systems)
+        trn_patterns = training_dataset_params.get("rglob_patterns", None)
+        training_systems = process_systems(training_systems, patterns=trn_patterns)
```

Comment:
`training_data/systems` is also used below
https://github.com/deepmodeling/deepmd-kit/blob/9b704878fa360069f9d04b2378044c51f855ab25/deepmd/pt/entrypoints/main.py#L313-L315